### PR TITLE
Add ember-lint

### PR DESCRIPTION
**Pull request by cadeParade**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -40,6 +40,11 @@ module.exports = {
       rules: {
         "no-unused-expressions": 'off'
       }
+    }, {
+      files: ['mirage/**'],
+      rules: {
+        "ember/avoid-leaking-state-in-ember-objects": 0
```
Comment (maprules1000) [inline .eslintrc.js]: Do you still need to ignore this rule if you ignore the whole mirage directory?

File: app/components/comparison-switcher-item.js [hunk 2]
```diff
@@ -13,6 +12,10 @@ export default Component.extend({
     'matchingComparison.isDifferent:has-diffs',
   ],
 
+  comparisons: computed(function() {
+    return [];
```
Comment (maprules1000) [inline app/components/comparison-switcher-item.js]: I don't know how I feel about creating a computed property that becomes not a computed property. Any reason for this pattern over the `init` one in the [ember eslint docs](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/avoid-leaking-state-in-ember-objects.md#description)?

File: app/components/form-fields/checkbox.js [hunk 3]
```diff
@@ -18,11 +18,6 @@ export default Component.extend({
     return this.get('changeset.' + this.get('property')) === this.get('checkedValue');
   }),
 
-  // Update the checked property if the changeset changes.
-  _updateChecked: observer('shouldBeChecked', function() {
```
Comment (maprules1000) [inline app/components/form-fields/checkbox.js]: 🙌

File: tests/unit/routes/application-test.js [hunk 4]
```diff
@@ -8,23 +8,11 @@ import {setupTest} from 'ember-mocha';
...
-      'service:auth0',
-      'service:analytics',
-      'service:raven',
-      'service:launch-darkly',
-      'service:headTags',
-      'service:redirects',
-      'service:intercom',
-    ],
-  });
+  setupTest();
```
Comment (maprules1000) [inline tests/unit/routes/application-test.js]: 🙌 This is sooooo much cleaner

File: .eslintrc.js (hunk 1 above)
Comment (cadeParade) [reply .eslintrc.js]: This ignores that specific rule, only in the mirage directory.

File: app/components/comparison-switcher-item.js (hunk 2 above)
Comment (cadeParade) [reply app/components/comparison-switcher-item.js]: I just hate init, but I don't have good reasons...I'll change it.

File: app/components/forms/base.js [hunk 5]
```diff
@@ -25,10 +24,11 @@ export default Component.extend({
 
     return new Changeset(model, lookupValidator(validator), validator);
   }),
-  focusOnInsert: on('didInsertElement', function() {
+
+  didInsertElement() {
```
Comment (maprules1000) [inline app/components/forms/base.js]: Maybe this will clean up some of our flakey focus diffs 🤔

File: app/components/snapshot-viewer-full.js [hunk 6]
```diff
@@ -35,7 +35,9 @@ export default Component.extend({
   }),
   selectedComparison: alias('filteredComparisons.selectedComparison'),
 
-  galleryMap: ['base', 'diff', 'head'],
+  galleryMap: computed(function() {
```
Comment (maprules1000) [inline app/components/snapshot-viewer-full.js]: Could you set this up in `init()`?

File: app/components/marketing/pricing-card-block.js [hunk 7]
```diff
@@ -20,10 +20,12 @@ export default Component.extend({
   attributeBindings: ['data-test-pricing-card-block'],
   'data-test-pricing-card-block': true,
 
-  essentialPlanSliderSetup: {
-    maxRange: 150000, // This is how many snapshots it takes to get slightly over $849
-    step: 5000, // arbitrary
-  },
+  essentialPlanSliderSetup: computed(function() {
```
Comment (maprules1000) [inline app/components/marketing/pricing-card-block.js]: This one too? I don't have very strong feels about it tbh

File: app/components/snapshot-viewer-full.js (hunk 6 above)
Comment (cadeParade) [reply app/components/snapshot-viewer-full.js]: Ok, but it never changes :P

File: app/components/marketing/pricing-card-block.js (hunk 7 above)
Comment (cadeParade) [reply app/components/marketing/pricing-card-block.js]: Also never changes but sure :)

**Pull request merged**